src/main/java/seedu/address/commons/core/Messages.java

```diff
@@ -10,4 +12,21 @@ public class Messages {
     public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
     public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
 
+    /**
+     * Formats the {@code person} for display to the user.
+     */
+    public static String format(Person person) {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(person.getName())
+                .append(" Phone: ")
+                .append(person.getPhone())
+                .append(" Email: ")
+                .append(person.getEmail())
+                .append(" Address: ")
+                .append(person.getAddress())
+                .append(" Tags: ");
+        person.getTags().forEach(builder::append);
```

Comment:
Example Output: John Doe Phone: 12345678 Email: johndoe@example.com Address: 123 Street Tags: [friend][classmate]

The Messages class serves as a utility class for formatting and storing user-visible messages. The class is utilized to create a user-friendly, readable string that represents a Person object. However, there are several aspects where this method could be improved:

1. Lack of Attribute Separators: The current formatting of the Person attributes doesn't include clear separators. This may lead to confusion when reading the output, particularly if a Person's name is erroneously interpreted as part of their phone number (for example, "John Doe Phone" might be misinterpreted as "John DoePhone"). Introducing distinct separators between attributes will greatly improve readability. A suggestion would be to include something like a colon (":") or a hyphen ("-") between the attributes.

2. Improving Code Cohesion: As it stands, the Messages class and ToStringBuilder class perform similar tasks, i.e., they both format objects into strings. To improve the code's cohesion, it might be beneficial to merge the Messages class's functionality into the ToStringBuilder class. This can help to keep all the string formatting logic in one place, making it easier for developers to find and use these APIs.

3. Enhancing Readability with ToStringBuilder: Since ToStringBuilder is already used in your codebase, it can be a good choice for implementing the format method, as it's designed for creating readable string representations of objects. The ToStringBuilder could be used to build the string with the added benefit of clear separation of fields, making the output easier to read.

Reply:
Regarding 2.
I appreciate the suggestion, but I have plans in a future commit/pr to move the `Messages` class to within `logic` package. As it currently stands, all functionalities of `Messages` class is related to displaying messages directly to the user via the GUI and it is right now exclusively used by the `logic` package. Thus, I feel that it is more appropriate to move the `Messages` class directly into the `logic` package instead of keeping it in `commons` package. 

The purpose of `ToStringBuilder` is mainly for devs to easily identify object string representations and their related data in logs and unit testings etc. Thus, it mainly serves as a way to enable easy debugging and testing. To enable this, user-facing messages will thus need to be migrated to use a different formatting method which was what was implemented here. 

However, I will definitely experiment adding additional purpose/functionalities based on your suggestion!

Reply:
For 1, isn't the code already using `;` as the attribute separator?

Reply:
> For 1, isn't the code already using `;` as the attribute separator?

The string output from the code when tested is as follows:

"John Doe Phone: 12345678 Email: johndoe@example.com Address: 123 Street Tags: [friend][classmate]"

This output is used for user interface display. The separators between different elements (name, phone, email, address, tags) are crucial for readability and user understanding.

Consider the following examples:

"John Doe Phone" or "Address: 123 Street Tags: [friend][classmate]"

Without proper separators, the information can blend together and become confusing for the user. Therefore, I feel it's important to ensure that the separators are clearly defined and consistently used throughout the output.

Reply:
My apologies, I forgot to post a new iteration via CIHR sorry for the confusion, I have manually added the separator `;` in the format.

Regarding adding multiple use case to the `toStringBuilder`, while it is possible I feel that the general use case for it might not as widely use. As mentioned before `toStringBuilder` was originally meant as an easy way to print out object string representation while formatting is related to UI side details. So after some tweaking of code, I feel that there isn't enough usage to justify expanding `toStringBuilder` to allow customizing its display format. 
Perhaps it would be better to reconsider this when there are more use cases in the future which require similar formatting for user side message.

Reply:
> hmm... `toString()` is usually used for internal purposes (such as logging and debugging), to produce a compact string representation of the object. While it _can_ be used for UI purposes, it might not be ideal as UI display have different considerations (easier reading, nicer formatting etc.). Is AB3 using `toString()` method of Person class for UI display?

Just realized that this method is specifically if for showring Person object details on the UI. So, the above doesn't apply.